datahub-web-react/src/app/settings/posts/PostsList.tsx

```diff
@@ -0,0 +1,199 @@
+import { addToListPostCache, removeFromListPostCache } from './utils';
+import { Message } from '../../shared/Message';
+import TabToolbar from '../../entity/shared/components/styled/TabToolbar';
+import { SearchBar } from '../../search/SearchBar';
+import { StyledTable } from '../../entity/shared/components/styled/StyledTable';
+
+const PostsContainer = styled.div``;
+
+export const PostsPaginationContainer = styled.div`
+    display: flex;
+    justify-content: center;
+    padding: 12px;
+    padding-left: 16px;
+    border-bottom: 1px solid;
+    border-color: ${(props) => props.theme.styles['border-color-base']};
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+`;
+
+const PaginationInfo = styled(Typography.Text)`
+    padding: 0px;
+`;
+
+const DEFAULT_PAGE_SIZE = 10;
+
+export const PostList = () => {
+    const entityRegistry = useEntityRegistry();
+    const location = useLocation();
+    const params = QueryString.parse(location.search, { arrayFormat: 'comma' });
+    const paramsQuery = (params?.query as string) || undefined;
+    const [query, setQuery] = useState<undefined | string>(undefined);
+    useEffect(() => setQuery(paramsQuery), [paramsQuery]);
+
+    const [page, setPage] = useState(1);
+    const [isCreatingPost, setIsCreatingPost] = useState(false);
+
+    const pageSize = DEFAULT_PAGE_SIZE;
+    const start = (page - 1) * pageSize;
+
+    const { loading, error, data, client, refetch } = useListPostsQuery({
+        variables: {
+            input: {
+                start,
+                count: pageSize,
+                query,
+            },
+        },
+        fetchPolicy: query && query.length > 0 ? 'no-cache' : 'cache-first',
+    });
+
+    const totalPosts = data?.listPosts?.total || 0;
+    const lastResultIndex = start + pageSize > totalPosts ? totalPosts : start + pageSize;
+    const posts = data?.listPosts?.posts || [];
+
+    const onChangePage = (newPage: number) => {
+        scrollToTop();
+        setPage(newPage);
+    };
+
+    const handleDelete = (urn: string) => {
+        removeFromListPostCache(client, urn, page, pageSize);
+        setTimeout(() => {
+            refetch?.();
+        }, 2000);
+    };
+
+    const allColumns = [
+        {
+            title: 'Title',
+            dataIndex: '',
+            key: 'title',
+            sorter: (sourceA, sourceB) => {
+                return sourceA.title.localeCompare(sourceB.title);
+            },
+            render: (record: PostEntry) => PostColumn(record.title, 200),
+            width: '20%',
+        },
+        {
+            title: 'Description',
+            dataIndex: '',
+            key: 'description',
+            render: (record: PostEntry) => PostColumn(record.description || ''),
+        },
+        {
+            title: 'Type',
+            dataIndex: '',
+            key: 'type',
+            render: (record: PostEntry) => PostColumn(record.contentType),
```

Comment:
reminder: we should make sure the copy shown aligns with what the user created (Announcement vs Link)

Reply:
definitely good call will update this